playwright/_impl/_network.py

```diff
@@ -247,6 +261,19 @@ async def continue_route() -> None:
 
         asyncio.create_task(continue_route())
 
+    async def _race_with_page_close(self, future: Coroutine) -> None:
+        if hasattr(self.request.frame, "_page"):
```

Comment:
why the check?

Reply:
Some dialog frame tests don't have a page super early on attached, we check it upstream too and the tests were failing otherwise.